src/VirtoCommerce.CatalogModule.Web/Scripts/blades/item-associations-list.js

```diff
@@ -1,160 +1,162 @@
-            template: 'Modules/$(VirtoCommerce.Catalog)/Scripts/blades/item-association-detail.tpl.html'
-        };
-        bladeNavigationService.showBlade(newBlade, blade);
-    };
-
-    function openAddEntityWizard() {
-        var newBlade = {
-            id: "associationWizard",
-            catalog: blade.catalog,
-            item: blade.item,
-            onSaveChanges: function () {
-                populateProductCatalogs(blade.item.associations);
-            },
-            controller: 'virtoCommerce.catalogModule.associationWizardController',
-            template: 'Modules/$(VirtoCommerce.Catalog)/Scripts/wizards/newAssociation/association-wizard.tpl.html'
-        };
-        bladeNavigationService.showBlade(newBlade, blade);
-    }
-
-    blade.toolbarCommands = [
-        {
-            name: "platform.commands.add", icon: 'fas fa-plus',
-            executeMethod: function () {
-                openAddEntityWizard();
-            },
-            canExecuteMethod: function () {
-                return true;
-            }
-        },
-        {
-            name: "platform.commands.delete", icon: 'fas fa-trash-alt',
-            executeMethod: function () {
-                $scope.deleteList($scope.gridApi.selection.getSelectedRows());
-            },
-            canExecuteMethod: function () {
-                return $scope.gridApi && _.any($scope.gridApi.selection.getSelectedRows());
-            }
-        }
-    ];
-
-    // ui-grid
-    $scope.setGridOptions = function (gridOptions) {
-        uiGridHelper.initialize($scope, gridOptions,
-            function (gridApi) {
-                // gridApi.grid.registerRowsProcessor($scope.singleFilter, 90);
-                gridApi.draggableRows.on.rowFinishDrag($scope, function () {
-                    for (var i = 0; i < blade.item.associations.length; i++) {
-                        blade.item.associations[i].priority = i + 1;
-                    }
-                });
-            });
-    };
-
-    initialize(blade.item);
-}]);
+angular.module('virtoCommerce.catalogModule')
+    .controller('virtoCommerce.catalogModule.itemAssociationsListController', ['$scope', 'platformWebApp.bladeNavigationService', '$timeout', 'uiGridConstants', 'platformWebApp.uiGridHelper', 'virtoCommerce.catalogModule.catalogs', 'virtoCommerce.catalogModule.items', 'virtoCommerce.catalogModule.categories', function ($scope, bladeNavigationService, $timeout, uiGridConstants, uiGridHelper, catalogs, items, categories) {
+        $scope.uiGridConstants = uiGridConstants;
+        var allCatalogIds = [];
+        var blade = $scope.blade;
+        blade.subtitle = 'catalog.widgets.itemAssociations.blade-subtitle';
+
+        blade.isLoading = false;
+        blade.refresh = function (item) {
+            initialize(item);
+        };
+
+        function initialize(item) {
+            blade.title = item.name;
+            blade.item = null;
+            $timeout(() => {

```

Comment:
add a comment for why we need to use $timeout here

Reply:
added